### PR TITLE
[plugin-excel] Add ability to add sheet to excel file

### LINKED ISSUE #2953: Add ability to generate tabs (sheets) in Excel file plugin
**Is your feature request related to a problem? Please describe.**
Current implementation of excel file generation give possibility to create a table
`When I create temporary excel file with content:$content and put path to $scopes variable `$variableName``

I'd like to have possibility to:
1. Generate file with multiple tabs and set names to each of them.
2. Fill each tab with table

### PR DESCRIPTION
**Pull request by uarlouski**
Closes #2953

### CODE REVIEW

File: vividus-tests/src/main/resources/story/integration/ExcelSteps.story [hunk 1]
```diff
@@ -57,15 +57,21 @@ When I initialize scenario variable `excel-data` with values:
...
+Scenario: Steps verification 'When I create temporary excel file containing sheet with name `$sheetName` and content:$content and put its path to $scopes variable `$variableName`' and 'When I add sheet with name `$sheetName` and content $content to excel file at path `$path`'
 Meta:
     @requirementId 2953
 When I create temporary excel file containing sheet with name `my-sheet-name` and content:
 |key1  |key2  |
 |value1|value2|
 and put its path to scenario variable `path`
+When I add sheet with name `added-sheet` and content:
+|key1  |key2  |
+|value1|value2|
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ExcelSteps.story:69]: let's use different data sets for different sheets

File: docs/modules/plugins/pages/plugin-excel.adoc [hunk 2]
```diff
@@ -128,6 +128,30 @@ When I create temporary excel file containing sheet with name `movies` and conte
 and put its path to scenario variable `movies-path`
 ----
 
+=== Add sheet to existing excel file
+
+Add new sheet with the specified name and content to the existing excel file.
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-excel.adoc:133]: What will happen if the sheet already exists?

Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-excel.adoc:133]: discussed

File: vividus-tests/src/main/resources/story/integration/ExcelSteps.story [hunk 3]
```diff
@@ -67,5 +67,20 @@ and put its path to scenario variable `path`
 When I initialize scenario variable `excel-data` with values:
 |key1  |key2  |
 |value1|value2|
+
+When I add sheet with name `added-sheet` and content:
+|animal             |weigh  |
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ExcelSteps.story:72]:
```suggestion
|animal             |weight  |
```

File: vividus-tests/src/main/resources/story/integration/ExcelSteps.story [hunk 4]
```diff
@@ -67,5 +67,20 @@ and put its path to scenario variable `path`
...
+|Komodo Dragon      |91 kg  |
+|Whale Shark        |21 ton |
+|Saltwater Crocodile|1075 kg|
+to excel file at path `${path}`
+When I initialize scenario variable `added-sheet-excel-data` with values:
+|animal             |weigh  |
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/ExcelSteps.story:78]:
```suggestion
|animal             |weight  |
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: vividus-plugin-excel/src/main/java/org/vividus/steps/ExcelFileSteps.java [hunk 5]
```diff
@@ -88,4 +88,17 @@ public void createExcelFileContainigSheetWithNameAndContent(String sheetName, Ex
...
+
+    /**
+     * Adds a new sheet with the specified name and content to the existing excel file specified by path.
+     * @param sheetName The name of the sheet
+     * @param content Any valid ExamplesTable that will be written to the sheet
+     * @param path The path to an exiting excel file
+     * @throws IOException if an I/O exception of some sort has occurred
+     */
+    @When("I add sheet with name `$sheetName` and content:$content to excel file at path `$path`")
+    public void addSheetToExcelFile(String sheetName, ExamplesTable content, Path path) throws IOException
```
[the remark this answers is not in the document]
Comment (ikalinin1) [reply vividus-plugin-excel/src/main/java/org/vividus/steps/ExcelFileSteps.java:100]: I just tested
image

**Pull request merged**